packages/components/upload/src/upload-content.vue

```diff
@@ -1,7 +1,12 @@
 <template>
   <div
-    :class="[ns.b(), ns.m(listType), ns.is('drag', drag)]"
-    tabindex="0"
+    :class="[
+      ns.b(),
+      ns.m(listType),
+      ns.is('drag', drag),
+      ns.is('disabled', disabled),
+    ]"
+    :tabindex="disabled ? '-1' : '0'"
```

Comment:
You also need to add the `disabled` to [L26](https://github.com/element-plus/element-plus/pull/17753/files#diff-848fa2c711dee63acb2665d4ec3ba52674c4644d2e2657bb123d2f0d613a053eR26)